src/server/pfs/fuse/filesystem.go

```diff
@@ -472,6 +472,9 @@ func (d *directory) readFiles(ctx context.Context) ([]fuse.Dirent, error) {
 		d.File.Path,
 		d.fs.getFromCommitID(d.File.Commit.Repo.Name),
 		d.Shard,
+		// setting recurse to false for performance reasons
+		// it does however means that we won't know the correct sizes of directories
+		false,
```

Comment:
It's fine to not know the correct sizes of directories, `ls` doesn't report the size of files stored in directories for normal filesystems either. And generally speaking the size of directory shown by `ls` isn't normally taken to be the size of the objects contained in it.